src/apps/schema/src/app/main.js

```diff
@@ -1,30 +1,44 @@
-import { connect } from "react-redux";
+import { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { Switch, Route, useRouteMatch } from "react-router-dom";
-
 import { WithLoader } from "@zesty-io/core/WithLoader";
 
 import { SchemaNav } from "./components/Nav";
 import { SchemaCreate } from "./views/SchemaCreate";
 import { SchemaEdit } from "./views/SchemaEdit";
 import { GettingStarted } from "./views/GettingStarted";
 
+import { fetchModels } from "shell/store/models";
+import { fetchSettings } from "shell/store/settings";
+
 import styles from "./main.less";
-export default connect((state) => {
-  return { navSchema: state.navSchema };
-})(function SchemaBuilder(props) {
+export default function SchemaBuilder() {
   const match = useRouteMatch("/schema/start");
   const showNav = match && match.isExact;
+  const navSchema = useSelector((state) => state.navSchema);
+  const dispatch = useDispatch();
+
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
+    Promise.all([dispatch(fetchModels()), dispatch(fetchSettings())]).then(
```

Comment:
Should we include a `catch` clause to handle if an API request fails?

Reply:
Sure. What is our standard to handle API failures? Can you point me to an existing case to base myself off on? Should we also add some to the content-editor .then chain?

Reply:
This is probably a good example: https://github.com/zesty-io/manager-ui/blob/master/src/apps/audit-trail/src/views/app/index.js#L26

In an api request failure we'd want to stop the loading state and show an error notification

Reply:
Sounds good. On it